templates/glossary.md

```diff
@@ -0,0 +1,331 @@
+
+The `@[simp]` attribute, for example, tags an object (ostensibly a `lemma` or `theorem`) as being a [simp lemma](#simp-lemma).
+
+For further details, including a list of commonly used attributes, see [this section of the Lean reference manual](https://leanprover.github.io/reference/other_commands.html#attributes).
+
+### beta reduction
+
+A specific simplification operation in [dependent type theory](#dependent-type-theory) (and Lean's implementation of it) which may be performed as part of deciding whether two [terms](#term) are [definitionally equal](#defeq).
+
+More precisely, it is the process of simplifying an expression such as `(λ x, x) a` by replacing appearances of the variable `x` with `a`.
+
+#### See also
+
+* [Section 2.3 of Theorem Proving in Lean](https://leanprover.github.io/theorem_proving_in_lean/dependent_type_theory.html#function-abstraction-and-evaluation)
+
+### big operators
+
+A [locale](#locale) within [mathlib](#mathlib)'s [algebra library](https://leanprover-community.github.io/mathlib_docs/algebra/big_operators/basic.html), enabled via `open_locale big_operators`.
+
+It defines notation for finite sums and products using the `∑` and `∏` characters.
+
+### binder
+
+Expressions such as `(a : α)`, `[a : α]` or `{a : α}` for any identifier(s) `a` and type `α` which, as part of various Lean syntactical elements -- [declarations](#declaration), `fun`, quantifiers and others -- represent identifiers which will be bound within the body of the syntactic element or declaration.
+
+Each type of binder has different implications for whether it will be bound implicitly (without being passed by a caller), explicitly, or via [typeclass inference](#typeclass-inference).
+
+In some places, notably within a `def`, defining "simple" binders without surrounding brackets are allowed, such as a binder `a` (with no explicit type) for some identifier `a`.
+
+### bundled vs unbundled
+
+Given a mathematical object `O` with property `P`, bundling `P` refers to the creation of a Lean [structure](#structure) containing a proof of `P` as one of its fields in addition to those needed to structurally define `O`.
+
+In contrast, an unbundled structure instead contains only the definition of `O`, and separately the creation of an `is_P` proposition which can be applied to terms of `O`.
+
+As a concrete example, a [group homomorphism](https://en.wikipedia.org/wiki/Group_homomorphism) can be viewed as a map `φ: G → H` between groups, along with a proof `h : φ(a * b) = φ(a) * φ(b)`. A bundled group homomorphism would contain both `φ` and `h` as fields, whereas an unbundled one would instead consist only of `φ`, with a separate `is_group_homomorphism` [declaration](#declaration) for the proof of `h`.
+
+There are performance, stylistic or implementation-related reasons to prefer bundling or unbundling.
+Generally within [mathlib](#mathlib), bundled structures are preferred, but unbundled versions are often also present.
```

Comment:
I'm unsure about this line, which doesn't exactly contradict 4.1 of https://arxiv.org/pdf/1910.09336.pdf , but also doesn't exactly agree. It might be good to reference this section of the paper here too.